devicetypes/smartthings/zwave-dual-switch.src/zwave-dual-switch.groovy

```diff
@@ -88,22 +95,32 @@ def parse(String description) {
 	return createEvent(result)
 }
 
-def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
-	[name: "switch", value: cmd.value ? "on" : "off"]
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, endpoint=null) {
+	(endpoint == 1) ? [name: "switch", value: cmd.value ? "on" : "off"] : [:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd, endpoint=null) {
+	(endpoint == 1) ? [name: "switch", value: cmd.value ? "on" : "off"] : [:]
 }
 
-def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
-	[name: "switch", value: cmd.value ? "on" : "off"]
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd, endpoint=null) {
+	(endpoint == 1) ? [name: "switch", value: cmd.value ? "on" : "off"] : [:]
 }
 
-def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
-	[name: "switch", value: cmd.value ? "on" : "off"]
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
 	def encapsulatedCommand = cmd.encapsulatedCommand([0x32: 3, 0x25: 1, 0x20: 1])
 	if (cmd.sourceEndPoint == 1) {
-		zwaveEvent(encapsulatedCommand)
+		zwaveEvent(encapsulatedCommand, 1)
```

Comment:
Trying to understand why this (and all the signatures) was changed?

Reply:
Each time the child device was turned on/off, the switch would generate a BasicReport that wasn't associated with either endpoint. This caused, for example, endpoint 1 to flip state when endpoint 2 was actuated. Since we're only interested in events from the endpoints, we filter out the non-associated ones.
There are probably a few ways to go about solving that problem, but this was the quickest. It also mimics how it's done in the Innovelli dual switch DTH.